main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -1052,6 +1005,68 @@ object Simplifier {
       SimplifiedAst.Op(sym, ann, mod, fparams, retTpe, eff, loc)
   }
 
+  private def visitStructNew(sym: Symbol.StructSym, givenFields: List[Symbol.StructFieldSym], regExpOpt: Option[SimplifiedAst.Expr], fieldExps: List[SimplifiedAst.Expr], tpe: Type, loc: SourceLocation)(implicit root: MonoAst.Root, flix: Flix): SimplifiedAst.Expr = {
```

Comment:
Now that this is a function - which is good - let us give it some documentation and a code example. Thanks!

Reply:
Probably most important to document the behavior of regExpOpt